--- src/osx/core/display.cpp.orig	2024-05-14 00:22:43
+++ src/osx/core/display.cpp	2024-07-23 08:01:53
@@ -30,6 +30,8 @@
 #endif
 
 #include "wx/osx/private.h"
+
+#ifdef __WXMAC__
 
 // ----------------------------------------------------------------------------
 // common helpers compiled even in wxUSE_DISPLAY==0 case
@@ -320,7 +322,7 @@
         depth = 16;
     else if ( CFStringCompare( pixelEncoding, CFSTR(IO8BitIndexedPixels) , kCFCompareCaseInsensitive) == kCFCompareEqualTo )
         depth = 8;
-    
+
     return depth;
 }
 
@@ -329,17 +331,17 @@
     wxArrayVideoModes resultModes;
 
     wxCFRef<CFArrayRef> theArray(CGDisplayCopyAllDisplayModes( m_id ,NULL ) );
-    
+
     for (CFIndex i = 0; i < CFArrayGetCount(theArray); ++i)
     {
         CGDisplayModeRef theValue = static_cast<CGDisplayModeRef>(const_cast<void*>(CFArrayGetValueAtIndex(theArray, i)));
-        
+
         wxVideoMode theMode(
                             CGDisplayModeGetWidth(theValue),
                             CGDisplayModeGetHeight(theValue),
                             wxOSXCGDisplayModeGetBitsPerPixel(theValue),
                             int(CGDisplayModeGetRefreshRate(theValue)));
-        
+
         if (theMode.Matches( mode ))
             resultModes.Add( theMode );
     }
@@ -350,7 +352,7 @@
 wxVideoMode wxDisplayImplMacOSX::GetCurrentMode() const
 {
     wxCFRef<CGDisplayModeRef> theValue( CGDisplayCopyDisplayMode( m_id ) );
-    
+
     return wxVideoMode(
                        CGDisplayModeGetWidth(theValue),
                        CGDisplayModeGetHeight(theValue),
@@ -370,20 +372,20 @@
 
     wxCHECK_MSG( mode.GetWidth() && mode.GetHeight(), false,
                 wxT("at least the width and height must be specified") );
-    
+
     bool bOK = false;
     wxCFRef<CFArrayRef> theArray(CGDisplayCopyAllDisplayModes( m_id ,NULL ) );
-    
+
     for (CFIndex i = 0; i < CFArrayGetCount(theArray); ++i)
     {
         CGDisplayModeRef theValue = static_cast<CGDisplayModeRef>(const_cast<void*>(CFArrayGetValueAtIndex(theArray, i)));
-        
+
         wxVideoMode theMode(
                             CGDisplayModeGetWidth(theValue),
                             CGDisplayModeGetHeight(theValue),
                             wxOSXCGDisplayModeGetBitsPerPixel(theValue),
                             int(CGDisplayModeGetRefreshRate(theValue)));
-        
+
         if ( theMode.GetWidth() == mode.GetWidth() && theMode.GetHeight() == mode.GetHeight() &&
             ( mode.GetDepth() == 0 || theMode.GetDepth() == mode.GetDepth() ) &&
             ( mode.GetRefresh() == 0 || theMode.GetRefresh() == mode.GetRefresh() ) )
@@ -393,7 +395,7 @@
             break;
         }
     }
-    
+
     return bOK;
 }
 
@@ -441,4 +443,13 @@
     return new wxDisplayFactorySingleMacOSX;
 }
 
+#else
+
+/* static */ wxDisplayFactory *wxDisplay::CreateFactory()
+{
+    return new wxDisplayFactorySingle;
+}
+
+#endif
+
 #endif // wxUSE_DISPLAY
