--- src/common/intl.cpp.orig	2024-05-14 00:22:43
+++ src/common/intl.cpp	2024-07-23 02:54:43
@@ -57,7 +57,7 @@
 
 #ifdef __WIN32__
     #include "wx/msw/private/uilocale.h"
-#elif defined(__WXOSX__)
+#elif defined(__APPLE__)
     #include "wx/osx/core/cfref.h"
     #include "wx/osx/core/cfstring.h"
     #include <CoreFoundation/CFLocale.h>
@@ -556,7 +556,7 @@
     default:
         encname.Printf(wxS("windows-%u"), codepage);
     }
-#elif defined(__WXMAC__)
+#elif defined(__APPLE__)
     encname = wxCFStringRef::AsString(
         CFStringGetNameOfEncoding(CFStringGetSystemEncoding())
     );
@@ -867,7 +867,7 @@
 // accessors for locale-dependent data
 // ----------------------------------------------------------------------------
 
-#if defined(__WINDOWS__) || defined(__WXOSX__)
+#if defined(__WINDOWS__) || defined(__APPLE__)
 
 namespace
 {
@@ -1382,7 +1382,7 @@
     return wxUILocale::GetCurrent().GetInfo(index, cat);
 }
 
-#elif defined(__WXOSX__)
+#elif defined(__APPLE__)
 
 // This function is also used by wxUILocaleImpl, so don't make it private.
 extern wxString
