From 61986f615b021b2c915e1ca2b2c43827d7ac6586 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 22 Jul 2024 18:25:28 +0800
Subject: [PATCH 2/5] mimetype.cpp: fix for macOS

---
 src/osx/core/mimetype.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git src/osx/core/mimetype.cpp src/osx/core/mimetype.cpp
index e60f105a4f..30458d98f6 100644
--- src/osx/core/mimetype.cpp
+++ src/osx/core/mimetype.cpp
@@ -17,7 +17,9 @@
 #if wxUSE_MIMETYPE
 
 #include "wx/osx/mimetype.h"
-#include "wx/osx/private.h"
+#include <CoreFoundation/CoreFoundation.h>
+#include <Carbon/Carbon.h>
+#include "wx/osx/core/cfstring.h"
 
 /////////////////////////////////////////////////////////////////////////////
 // Helper functions
-- 
2.45.2

