From cf2bdd9999c79d94c5469ed69bc8b71a14c94efb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 22 Jul 2024 21:08:36 +0800
Subject: [PATCH 5/5] Fix cfstring

---
 include/wx/osx/core/cfstring.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git include/wx/osx/core/cfstring.h include/wx/osx/core/cfstring.h
index e1a557fde8..4ff4590de3 100644
--- include/wx/osx/core/cfstring.h
+++ include/wx/osx/core/cfstring.h
@@ -41,7 +41,7 @@
     wxCFStringRef(const wxString &str,
                         wxFontEncoding encoding = wxFONTENCODING_DEFAULT) ;
 
-#ifdef __OBJC__
+#ifdef __WXMAC__
     wxCFStringRef(WX_NSString ref)
         : wxCFRef< CFStringRef >((WX_OSX_BRIDGE_RETAINED CFStringRef) ref)
     {
@@ -69,8 +69,7 @@
 #ifdef __WXMAC__
     static wxString AsString( WX_NSString ref, wxFontEncoding encoding = wxFONTENCODING_DEFAULT ) ;
     static wxString AsStringWithNormalizationFormC( WX_NSString ref, wxFontEncoding encoding = wxFONTENCODING_DEFAULT ) ;
-#endif
-#ifdef __OBJC__
+
     WX_NSString AsNSString() const { return (WX_OSX_BRIDGE WX_NSString)(CFStringRef) *this; }
 #endif
 private:
