--- Makefile.in.orig	2024-07-23 15:08:45
+++ Makefile.in	2024-07-23 15:11:08
@@ -2711,7 +2711,8 @@
 	wx/osx/core/cftype.h \
 	wx/osx/core/joystick.h \
 	wx/osx/core/mimetype.h \
-	wx/osx/core/dataview.h
+	wx/osx/core/dataview.h \
+	wx/osx/cocoa/stdpaths.h
 @COND_TOOLKIT_@BASE_OSX_HDR = $(COND_TOOLKIT__BASE_OSX_HDR)
 @COND_PLATFORM_MACOSX_1@BASE_PLATFORM_HDR = $(BASE_OSX_HDR)
 COND_PLATFORM_UNIX_1_BASE_PLATFORM_HDR =  \
@@ -3509,7 +3510,6 @@
 	wx/osx/cocoa/evtloop.h \
 	wx/osx/cocoa/private.h \
 	wx/generic/region.h \
-	wx/osx/cocoa/stdpaths.h \
 	wx/osx/sound.h \
 	wx/osx/dvrenderer.h \
 	wx/osx/cocoa/dataview.h \
