--- build/files.orig	2024-05-14 00:22:43
+++ build/files	2024-07-23 06:25:02
@@ -163,7 +163,6 @@
 
 # Base files used by OS X ports (not Carbon)
 BASE_OSX_SHARED_SRC =
-    src/osx/core/mimetype.cpp
     src/osx/fswatcher_fsevents.cpp
     src/osx/cocoa/stdpaths.mm
     $(BASE_COREFOUNDATION_SRC)
@@ -176,6 +175,7 @@
 
 # Base and GUI files used by OS X
 BASE_AND_GUI_OSX_COCOA_SRC =
+    src/osx/core/mimetype.cpp
     src/osx/cocoa/utils.mm
     src/osx/cocoa/power.mm
     src/osx/volume.mm
