From f45a6ba3f6b6a791749f818646495908ba013e56 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 2 Jan 2024 16:50:59 +0800
Subject: [PATCH 3/6] Remove wrong options from main CMakeLists

---
 CMakeLists.txt | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index 493a1b65..a33a326f 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -77,7 +77,7 @@ if (UNIX)
 
 	# warnings as errors:
 	# we have a problem with people checking in code with warnings.
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-unused-local-typedef")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
 
 	if (NOT APPLE)
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
@@ -166,23 +166,7 @@ if (UNIX)
 	endif()
 	
 	if (APPLE)
-		message(STATUS "OSX_TARGET_MAJOR=${OSX_TARGET_MAJOR}")
-		message(STATUS "OSX_TARGET_MINOR=${OSX_TARGET_MINOR}")
 
-		if (NOT (OSX_TARGET_MAJOR EQUAL 10))
-			message(FATAL_ERROR "Mac OS X target must be 10.x")
-		endif ()
-
-		if (OSX_TARGET_MINOR LESS 6)
-			# <= 10.5: 32-bit Intel and PowerPC
-			set(CMAKE_OSX_ARCHITECTURES "ppc;i386"
-				CACHE STRING "" FORCE)
-		else()
-			# >= 10.6: Intel only
-			set(CMAKE_OSX_ARCHITECTURES "i386"
-				CACHE STRING "" FORCE)
-		endif()
-		
 		set(CMAKE_CXX_FLAGS "--sysroot ${CMAKE_OSX_SYSROOT} ${CMAKE_CXX_FLAGS} -DGTEST_USE_OWN_TR1_TUPLE=1")
 
 		find_library(lib_ScreenSaver ScreenSaver)
-- 
2.43.0

