From 6bee253ca60515e297b7e6ed77fd852929eddf5c Mon Sep 17 00:00:00 2001
From: eyedeekay <idk@mulder>
Date: Wed, 7 Feb 2024 11:34:39 -0500
Subject: [PATCH 02/18] Use PID/TickCount for random nickname string on
 __APPLE__ Platforms

---
 src/libsam3a/libsam3a.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git src/libsam3a/libsam3a.c src/libsam3a/libsam3a.c
index a996ee6..e812f00 100644
--- src/libsam3a/libsam3a.c
+++ src/libsam3a/libsam3a.c
@@ -665,6 +665,7 @@ static uint32_t genSeed(void) {
   volatile uint32_t seed = 1;
   uint32_t res;
 #ifndef WIN32
+  #ifndef __APPLE__
   struct sysinfo sy;
   pid_t pid = getpid();
   //
@@ -672,6 +673,10 @@ static uint32_t genSeed(void) {
   res = hashint((uint32_t)pid) ^ hashint((uint32_t)time(NULL)) ^
         hashint((uint32_t)sy.sharedram) ^ hashint((uint32_t)sy.bufferram) ^
         hashint((uint32_t)sy.uptime);
+  #else
+  res = hashint((uint32_t)GetCurrentProcessId()) ^
+        hashint((uint32_t)GetTickCount());
+  #endif
 #else
   res = hashint((uint32_t)GetCurrentProcessId()) ^
         hashint((uint32_t)GetTickCount());
-- 
2.43.1

