From 390793bdb8abbf841b837d5c99c45d3307fa566d Mon Sep 17 00:00:00 2001
From: eyedeekay <idk@mulder>
Date: Thu, 8 Feb 2024 11:21:18 -0500
Subject: [PATCH 06/18] Use getpid and TickCount when on OSX platforms

---
 src/libsam3/libsam3.c | 22 +++++++++++++++-------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git src/libsam3/libsam3.c src/libsam3/libsam3.c
index 522bb82..a619955 100644
--- src/libsam3/libsam3.c
+++ src/libsam3/libsam3.c
@@ -631,19 +631,27 @@ static inline uint32_t hashint(uint32_t a) {
 }
 
 static uint32_t genSeed(void) {
+  volatile uint32_t seed = 1;
   uint32_t res;
 #ifndef WIN32
-  struct sysinfo sy;
-  pid_t pid = getpid();
-  //
-  sysinfo(&sy);
-  res = hashint((uint32_t)pid) ^ hashint((uint32_t)time(NULL)) ^
-        hashint((uint32_t)sy.sharedram) ^ hashint((uint32_t)sy.bufferram) ^
-        hashint((uint32_t)sy.uptime);
+  #ifndef __APPLE__
+    struct sysinfo sy;
+    pid_t pid = getpid();
+    //
+    sysinfo(&sy);
+    res = hashint((uint32_t)pid) ^ hashint((uint32_t)time(NULL)) ^
+          hashint((uint32_t)sy.sharedram) ^ hashint((uint32_t)sy.bufferram) ^
+          hashint((uint32_t)sy.uptime);
+  #else
+    res = hashint((uint32_t)getpid()) ^
+          hashint((uint32_t)TickCount());
+  #endif
 #else
   res = hashint((uint32_t)GetCurrentProcessId()) ^
         hashint((uint32_t)GetTickCount());
 #endif
+  res += __sync_fetch_and_add(&seed, 1);
+  //
   return hashint(res);
 }
 
-- 
2.43.1

