From b6d51e6284ab5aa42f549a697784400b64572dbc Mon Sep 17 00:00:00 2001
From: eyedeekay <idk@mulder>
Date: Tue, 13 Feb 2024 10:17:17 -0500
Subject: [PATCH 11/18] since SOL_SOCKET doesn't exist on Mac, pass SOL_TCP on
 that platform

---
 src/libsam3/libsam3.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git src/libsam3/libsam3.c src/libsam3/libsam3.c
index e126e24..48b150e 100644
--- src/libsam3/libsam3.c
+++ src/libsam3/libsam3.c
@@ -59,6 +59,9 @@
 
 #if defined(__unix__) && !defined(__APPLE__)
 #include <sys/sysinfo.h>
+int solSocket() {
+  return SOL_SOCKET;
+}
 #endif
 
 #if defined(__APPLE__)
@@ -68,6 +71,9 @@ uint32_t TickCount() {
   uint32_t mul = 0x80d9594e;
   return ((((0xffffffff & mat) * mul) >> 32) + (mat >> 32) * mul) >> 23;
 }
+int solSocket() {
+  return SOL_TCP;
+}
 #endif
 
 ////////////////////////////////////////////////////////////////////////////////
@@ -92,7 +98,7 @@ int sam3tcpSetTimeoutSend(int fd, int timeoutms) {
     struct timeval tv;
     //
     ms2timeval(&tv, timeoutms);
-    return (setsockopt(fd, SOL_SOCKET, SO_SNDTIMEO, &tv, sizeof(tv)) < 0 ? -1
+    return (setsockopt(fd, solSocket(), SO_SNDTIMEO, &tv, sizeof(tv)) < 0 ? -1
                                                                          : 0);
   }
   return -1;
@@ -103,7 +109,7 @@ int sam3tcpSetTimeoutReceive(int fd, int timeoutms) {
     struct timeval tv;
     //
     ms2timeval(&tv, timeoutms);
-    return (setsockopt(fd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv)) < 0 ? -1
+    return (setsockopt(fd, solSocket(), SO_RCVTIMEO, &tv, sizeof(tv)) < 0 ? -1
                                                                          : 0);
   }
   return -1;
@@ -144,7 +150,7 @@ int sam3tcpConnectIP(uint32_t ip, int port) {
     }
   }
   //
-  setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, &val, sizeof(val));
+  setsockopt(fd, solSocket(), SO_KEEPALIVE, &val, sizeof(val));
   //
   if (connect(fd, (struct sockaddr *)&addr, sizeof(struct sockaddr_in)) < 0) {
     if (libsam3_debug)
-- 
2.43.1

